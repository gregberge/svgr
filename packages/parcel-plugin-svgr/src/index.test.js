import Bundler from 'parcel-bundler'
import path from 'path'
import plugin from '.'

const getCode = bundle =>
  Array.from(bundle.assets)
    .filter(asset => asset.id === 'icon.svg')
    .map(asset => asset.generated)

describe('parcel plugin', () => {
  it('should convert file', async () => {
    const bundler = new Bundler(path.join(__dirname, '__fixtures__/icon.svg'), {
      outDir: path.join(__dirname, '../../../__fixtures_build__/parcel'),
      cache: false,
      detailedReport: false,
      hmr: false,
      logLevel: 0,
      target: 'node',
      watch: false,
    })

    plugin(bundler)

    const bundle = await bundler.bundle()
    const code = getCode(bundle)
    expect(code).toMatchSnapshot()
  })
})
