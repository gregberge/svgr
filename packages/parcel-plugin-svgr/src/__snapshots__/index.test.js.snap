// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parcel plugin should convert file 1`] = `
Array [
  Object {
    "js": "\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require(\\"react\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() {
  _extends = Object.assign || function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };

  return _extends.apply(this, arguments);
}

var _ref =
/*#__PURE__*/
_react.default.createElement(\\"g\\", {
  stroke: \\"#063855\\",
  strokeWidth: 2,
  fill: \\"none\\",
  fillRule: \\"evenodd\\",
  strokeLinecap: \\"square\\"
}, _react.default.createElement(\\"path\\", {
  d: \\"M51 37L37 51M51 51L37 37\\"
}));

var SvgIcon = function SvgIcon(props) {
  return _react.default.createElement(\\"svg\\", _extends({
    width: 88,
    height: 88
  }, props), _ref);
};

var _default = SvgIcon;
exports.default = _default;",
    "map": undefined,
  },
]
`;
